ante/fee.go

```diff
@@ -33,17 +48,34 @@ func (mfd BypassMinFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 	if !ok {
 		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
 	}
-
-	feeCoins := feeTx.GetFee()
+	// getFee only get non zero fees?
+	feeCoins := feeTx.GetFee().Sort()
 	gas := feeTx.GetGas()
 	msgs := feeTx.GetMsgs()
 
-	// Only check for minimum fees if the execution mode is CheckTx and the tx does
+	//todo if need to check gas > uint64(len(msgs))*maxBypassMinFeeMsgGasUsage) here ?
+
+	// Only check for minimum fees and global fee if the execution mode is CheckTx and the tx does
 	// not contain operator configured bypass messages. If the tx does contain
 	// operator configured bypass messages only, it's total gas must be less than
-	// or equal to a constant, otherwise minimum fees are checked to prevent spam.
+	// or equal to a constant, otherwise minimum fees and global fees are checked to prevent spam.
 	if ctx.IsCheckTx() && !simulate && !(mfd.bypassMinFeeMsgs(msgs) && gas <= uint64(len(msgs))*maxBypassMinFeeMsgGasUsage) {
```

Comment:
Could make this a bit more legible with variable names

```suggestion

	containsOnlyBypassMinFeeMsgs := mfd.bypassMinFeeMsgs(msgs)
	doesNotExceedMaxGasUsage := gas <= uint64(len(msgs))*maxBypassMinFeeMsgGasUsage
	allowedToBypassMinFee := containsOnlyBypassMinFeeMsgs && doesNotExceedMaxGasUsage

	if ctx.IsCheckTx() && !simulate && !allowedToBypassMinFee {
```